src/transformState.js

```diff
@@ -5,7 +5,25 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (let i = 0; i < actions.length; i++) {
+    const oneElement = actions[i];
+
+    if (oneElement.type === 'addProperties') {
+      Object.assign(state, oneElement.extraData);
+    } else if (oneElement.type === 'removeProperties') {
+      const keysRemoving = oneElement.keysToRemove;
+
+      for (let m = 0; m < keysRemoving.length; m++) {
+        delete state[`${keysRemoving[m]}`];
+      }
+    } else if (oneElement.type === 'clear') {
+      for (const key in state) {
+        delete state[key];
+      }
+    }
+  }
```

Comment:
Your solution works fine, good job!
But I really recommend to try to solve it with switch. For now you have 3-deep indentation. It makes code hard to read.